packages/lib/server/repository/organization.ts

```diff
@@ -126,7 +126,7 @@ export class OrganizationRepository {
           },
         },
         metadata: {
-          ...(IS_TEAM_BILLING_ENABLED ? { requestedSlug: orgData.slug } : {}),
+          ...(IS_TEAM_BILLING_ENABLED && !orgData.isPlatform ? { requestedSlug: orgData.slug } : {}),
```

Comment:
We should atleast consolidate these two variables in a logical variable name like 
`shouldSetRequestedSlug`
```suggestion
          ...(IS_TEAM_BILLING_ENABLED && !orgData.isPlatform ? { requestedSlug: orgData.slug } : {}),
```